bash.sh

```diff
@@ -53,14 +53,34 @@ array_filter() {
 #
 # Join array elements with a string.
 #
-# $1: String separator
-# $@: Array elements
+# $1: String default, used in case array to join is empty
```

Comment:
Based on the principle of [do one thing and do it well](https://en.wikipedia.org/wiki/Unix_philosophy#Do_One_Thing_and_Do_It_Well), can you remove the "default"?

I feel this is conflating two responsibilities into one function (`join` should just join, nothing more, nothing less).